code/modules/paperwork/paperbin.dm

```diff
@@ -120,50 +144,92 @@
 
 /obj/item/paper_bin/update_overlays()
 	. = ..()
+
+	total_paper = LAZYLEN(papers)
+
+	if(bin_pen)
+		pen_overlay = mutable_appearance(bin_pen.icon, bin_pen.icon_state)
+
+	if(!bin_overlay)
+		bin_overlay = mutable_appearance(icon, bin_overlay_string)
+
+	if(LAZYLEN(papers))
+		var/paper_number = 1
+		for(var/obj/item/paper/current_paper in papers)
+			if(paper_number == LAZYLEN(papers) || paper_number % 8 == 0)
+				var/mutable_appearance/paper_overlay = mutable_appearance(current_paper.icon, current_paper.icon_state)
+				paper_overlay.color = current_paper.color
+				paper_overlay.pixel_y = paper_number/8 - 2 //gives the illusion of stacking
+				. += paper_overlay
+				if(paper_number == LAZYLEN(papers)) //this is our top paper
+					. += current_paper.overlays //add overlays only for top paper
+					if(istype(src, /obj/item/paper_bin/bundlenatural))
+						bin_overlay.pixel_y = paper_overlay.pixel_y //keeps binding centred on stack
+					if(bin_pen)
+						pen_overlay.pixel_y = paper_overlay.pixel_y //keeps pen on top of stack
+			paper_number++
+		. += bin_overlay
+
 	if(bin_pen)
-		. += mutable_appearance(bin_pen.icon, bin_pen.icon_state)
+		. += pen_overlay
 
 /obj/item/paper_bin/construction
 	name = "construction paper bin"
 	desc = "Contains all the paper you'll never need, IN COLOR!"
-	icon_state = "paper_binc"
 	papertype = /obj/item/paper/construction
 
 /obj/item/paper_bin/bundlenatural
 	name = "natural paper bundle"
 	desc = "A bundle of paper created using traditional methods."
-	icon_state = "paper_bundle"
+	icon_state = null
 	papertype = /obj/item/paper/natural
 	resistance_flags = FLAMMABLE
+	bin_overlay_string = "paper_bundle_overlay"
+	///Cable this bundle is held together with.
+	var/obj/item/stack/cable_coil/binding_cable
+
+/obj/item/paper_bin/bundlenatural/Initialize(mapload)
+	binding_cable = new /obj/item/stack/cable_coil(src, 2)
+	binding_cable.color = "#a9734f"
+	binding_cable.cable_color = "brown"
+	binding_cable.desc += " Non-natural."
+	return ..()
+
+/obj/item/paper_bin/bundlenatural/dump_contents(atom/droppoint)
+	. = ..()
+	qdel(src)
+
+/obj/item/paper_bin/bundlenatural/update_overlays()
+	bin_overlay = mutable_appearance(icon, bin_overlay_string)
+	bin_overlay.color = binding_cable.color
+	return ..()
 
 /obj/item/paper_bin/bundlenatural/attack_hand(mob/user, list/modifiers)
 	..()
-	if(total_paper < 1)
-		qdel(src)
+	if(!LAZYLEN(papers))
+		deconstruct(FALSE)
+
+/obj/item/paper_bin/bundlenatural/deconstruct(disassembled)
+	dump_contents()
+	qdel(src)
 
 /obj/item/paper_bin/bundlenatural/fire_act(exposed_temperature, exposed_volume)
 	qdel(src)
 
 /obj/item/paper_bin/bundlenatural/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/paper/carbon) && (W.icon_state == "paper_stack" || W.icon_state == "paper_stack_words"))
+		to_chat(user, "<span class='warning'>[W] won't fit into [src].</span>")
+		return
```

Comment:
is there a way to achieve this without checking icon state names?

Reply:
I'd prefer to compare the icon sizes but I'm not sure how to do that

Reply:
yeah that's not possible with byond

Reply:
I just refused all carbon paper, even though when the top sheet has been removed they're the correct width. Ideally I'd transform it to be one pixel less in width but I don't know how to do that